# HG changeset patch
# User Kai T. Ohlhus <k.ohlhus@gmail.com>
# Date 1484348501 -3600
#      Sat Jan 14 00:01:41 2017 +0100
# Node ID d8d0e600cbc4ea5633a379255d2070fffd5d01c6
# Parent  28668bfd5b6391432dbd1ad666ae2b7f5bea3568
# Parent  0412dc701e28fb1dae76bb6369e825a34ad1caad
maint: merge stable to default.

diff --git a/libinterp/corefcn/mexproto.h b/libinterp/corefcn/mexproto.h
--- a/libinterp/corefcn/mexproto.h
+++ b/libinterp/corefcn/mexproto.h
@@ -303,34 +303,34 @@ extern OCTINTERP_API mwIndex mxCalcSingl
 
 extern OCTINTERP_API size_t mxGetElementSize (const mxArray *ptr);
 
 #if defined (MEX_DEBUG)
 
 #  define mxAssert(expr, msg)                                           \
   do                                                                    \
     {                                                                   \
-      if (! expr)                                                       \
+      if (! (expr))                                                     \
         {                                                               \
           if (msg != NULL && msg[0] != '\0')                            \
             mexErrMsgIdAndTxt ("Octave:MEX",                            \
                                "Assertion failed: %s, at line %d of file \"%s\".\n%s\n", \
                                #expr, __LINE__, __FILE__, msg);         \
           else                                                          \
             mexErrMsgIdAndTxt ("Octave:MEX",                            \
                                "Assertion failed: %s, at line %d of file \"%s\".\n", \
                                #expr, __LINE__, __FILE__);              \
         }                                                               \
     }                                                                   \
   while (0)
 
 #  define mxAssertS(expr, msg)                                          \
   do                                                                    \
     {                                                                   \
-      if (! expr)                                                       \
+      if (! (expr))                                                     \
         {                                                               \
           if (msg != NULL && msg[0] != '\0')                            \
             mexErrMsgIdAndTxt ("Octave:MEX",                            \
                                "Assertion failed at line %d of file \"%s\".\n%s\n", \
                                __LINE__, __FILE__, msg);                \
           else                                                          \
             mexErrMsgIdAndTxt ("Octave:MEX",                            \
                                "Assertion failed at line %d of file \"%s\".\n", \
@@ -344,9 +344,8 @@ extern OCTINTERP_API size_t mxGetElement
 #  define mxAssertS(expr, msg)
 #endif
 
 #if defined (__cplusplus)
 }
 #endif
 
 #endif
-
